### PR TITLE
[WIP] ResNet-18

### PR DESCRIPTION
**Pull request by OTapio**
Created ResNet-18 for Flux. It does function but still needs some work. Has issues with speed and memory consumption. Feedback would be appreciated.

### CODE REVIEW

**Comment by dhairyagandhi96** [comment]
Might be better as a PR to Metalhead, to make it capable of emitting common `Resnet`s The default is ResNet50, but it should be able to emit the kinds we may want.

File: vision/cifar10/resnet18.jl [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+  SkipConnection(convolution_layer(64), Combinator(64)),
+  SkipConnection(identity_layer(128), (variable_1, variable_2) -> variable_1 + variable_2),
+  SkipConnection(convolution_layer(128), Combinator(128)),
+  SkipConnection(identity_layer(256), (variable_1, variable_2) -> variable_1 + variable_2),
+  SkipConnection(convolution_layer(256), Combinator(256)),
+  SkipConnection(identity_layer(512), (variable_1, variable_2) -> variable_1 + variable_2),
+  MeanPool((7,7)),
+  x -> reshape(x, :, size(x,4)),
+  Dense(512*1, 10),
+  softmax,
```
Comment (CarloLucibello) [inline vision/cifar10/resnet18.jl:134]: you should remove the softmax here since you are using `logicrossentropy`

File: vision/cifar10/resnet18.jl [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+  l = length(data_set)*batch_size
+  s = 0f0
+  for (x,y) in data_set
+    s += sum((onecold(m(x|>gpu) |> cpu) .== onecold(y|>cpu)))
+  end
+  return s/l
+end
+
+
+loss(x, y) = sum(logitcrossentropy(m(x), y))
```
Comment (CarloLucibello) [inline vision/cifar10/resnet18.jl:150]: If I'm not wrong, `logitcrossentropy` already performs reduction, no need for sum

File: vision/cifar10/resnet18.jl [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  SkipConnection(identity_layer(64), (variable_1, variable_2) -> variable_1 + variable_2),
+  SkipConnection(identity_layer(64), (variable_1, variable_2) -> variable_1 + variable_2),
+  SkipConnection(convolution_layer(64), Combinator(64)),
+  SkipConnection(identity_layer(128), (variable_1, variable_2) -> variable_1 + variable_2),
+  SkipConnection(convolution_layer(128), Combinator(128)),
+  SkipConnection(identity_layer(256), (variable_1, variable_2) -> variable_1 + variable_2),
+  SkipConnection(convolution_layer(256), Combinator(256)),
+  SkipConnection(identity_layer(512), (variable_1, variable_2) -> variable_1 + variable_2),
+  MeanPool((7,7)),
+  x -> reshape(x, :, size(x,4)),
```
Comment (mcabbott) [inline vision/cifar10/resnet18.jl:132]: Don't these anonymous functions have names?
```suggestion
  SkipConnection(identity_layer(512), +),
  MeanPool((7,7)),
  Flux.flatten,

```
